internal/stack/shellinit/shell/code.go

```diff
@@ -0,0 +1,19 @@
+package shell
```

Comment:
I think this can go in 2 direction: 1 is removing the package, the other is making this more generic (so the template is not a string allows creating the init code through a builder pattern for example).

I'm moving everything in a single file to keep it simple (given the POSIX enhancement you proposed) as we don't need anything fancy at the moment.